### PR TITLE
Large file upload exception handling

### PR DESCRIPTION
**Pull request by 94rain**
## Describe your changes
* [x] Add exception handling for GET request in get_presigned_link and POST request in upload_zipped_project
* [x] Add `runid` as an argument of `upload_zipped_project` so that it can abort the run if there are exceptions
* [x] Add a test for `api.zip_project` when the file is too large (client-side check)
* [x] Add a test for `upload_zipped_project` (directly make a POST request).

## Issue ticket number and link
Part of ploomber/pcloud-backend#17

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/ploomber/cloud/api.py [hunk 1]
```diff
@@ -310,15 +314,32 @@ def zip_project(force, runid, github_number, verbose, ignore_prefixes=None):
 
 @auth_header
 def get_presigned_link(headers):
-    return _requests.get(f"{HOST}/upload", headers=headers).json()
+    try:
+        return _requests.get(f"{HOST}/upload", headers=headers).json()
+    except Exception as err:
+        return {"Error": err}
```
Comment (idomic) [inline src/ploomber/cloud/api.py:320]: Make the error more informative

File: tests/cloud/test_api.py [hunk 2]
```diff
@@ -61,6 +61,31 @@ def test_zip_project_ignore_prefixes(tmp_directory, sample_project):
...
+         pytest.raises(BaseException) as excinfo:
+        stat.return_value.st_size = 6 * 1024 * 1024
+        api.zip_project(force=False,
+                        runid='runid',
+                        github_number='number',
+                        verbose=False)
+    assert "Your project's source code is over 5MB" in str(excinfo.value)
+
+
+def test_upload_zipped_project_errors_if_too_large(tmp_directory):
```
Comment (idomic) [inline tests/cloud/test_api.py:75]: This will probably fail until we push the cloud update right?

**Review by idomic — CHANGES_REQUESTED** [review]
looking good small comments, please also remove this a file

File: tests/cloud/test_api.py (hunk 2 above)
Comment (94rain) [reply tests/cloud/test_api.py:75]: Yes

File: src/ploomber/cloud/api.py (hunk 1 above)
Comment (94rain) [reply src/ploomber/cloud/api.py]: Updated

**Comment by 94rain** [comment]
> Not planned.

Just curious, why is this no longer planned?

**Pull request closed without merging**